src/worker/core/phase/newPhaseBeforeDraft.ts

```diff
@@ -50,6 +50,46 @@ const newPhaseBeforeDraft = async (
 			await idb.cache.players.put(p);
 		}
 	}
+	const challengeLoseBestPlayer = g.get("challengeLoseBestPlayer");
+	if (challengeLoseBestPlayer) {
+		const tids = g.get("userTids");
+		for (let i = 0; i < tids.length; i++) {
+			const tid = tids[i];
+			const [playersAll] = await Promise.all([
+				idb.cache.players.indexGetAll("playersByTid", tid),
+			]);
+			let pid = 0;
+			let bestOv = 0;
+			let bestPlayer: any = undefined;
```

Comment:
`let bestPlayer: Player | undefined;` - no need to make it `any`. Then after the loop to find the player, do
`if (bestPlayer) {` and TypeScript will be smart enough to know that inside that condition `bestPlayer` is `Player`, not `undefined`. That also gets rid of the type cast on line 72. In general type casts are dangerous and should be avoided unless absolutely necessary... sometimes they are needed, but not here. Let TypeScript help you rather than fighting it :)